portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -61,11 +61,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     
       ArrayList<Comment> comments = new ArrayList<>();
       for (Entity entity : results.asList(options)) {
-          String name = getNickname(userService.getCurrentUser().getUserId());
           String content = (String) entity.getProperty("content");
           long timestamp = (long) entity.getProperty("timestamp");
+          String userId = (String) entity.getProperty("userId");
+          String name = (String) getNickname(userId);
+          long commentId = entity.getKey().getId();
 
-          Comment comment = new Comment(name, content, timestamp);
+          Comment comment = new Comment(name, content, timestamp, userId, commentId);
```

Comment:
This looks like it could just be a constructor for a comment that takes an Entity as an input since the logic is getting more complex.